Feat/#17 사진작가 마이페이지와 프로필 뷰 분리

Hamthoven opened the pull request:
## 작업한 내용
- 수정/문의 버튼 추가
- 마이페이지/프로필 분리
- UIFont+에 b14 추가
- 일반 사용자 UI에도 사용하기 위해 컴포넌트 분리


## PR Point
<!-- 피드백을 받고 싶은 부분, 공유하고 싶은 부분, 작업 과정, 이유를 적어주세요. -->
- 일반 사용자용 레이아웃 함수는 아직 제작하지 않은 게 맞고 이후 일반사용자용 뷰 만들 때 다시 쓸 예정
- 컴포넌트 프로퍼티도 마찬가지로 아직 사진작가용에만 맞게 구성되어있음 (안보이게 처리하는 함수도 나중에 구현)
- 수정/문의 버튼에 폰트 b14로 설정했는데 다른 방법들 해봐도 적용 안되길래 일단 기본적인 설정 방법으로 코드 작성
- 네비게이션 바 머지 전에 했던 탓에 아직 네비게이션 바 적용이 안되었음.


## 📸 스크린샷
<!-- gif or mp4 용량 제한이 있는데... 용량 넘어가면 슬랙으로 보내 주세요. -->
<img width="320" alt="image" src="https://github.com/2023-SWU-Snapfit/SNAPFIT-iOS/assets/86394389/84dcf198-6658-460d-a63f-86a44d2ec37a">
<img width="233" alt="image" src="https://github.com/2023-SWU-Snapfit/SNAPFIT-iOS/assets/86394389/736d10ac-7e21-486c-8e8a-4761f752d492">


## 관련 이슈
- Resolved: #17 


<!-- Assignee, Reviewer 설정! 😇 -->

Hamthoven committed:
캐러셀 Class이름 오타 수정

Hamthoven committed:
#17 UIFont+에 b14 추가

Hamthoven committed:
마이페이지, 프로필 페이지 분리 & 문의/수정 버튼 추가

Hamthoven committed:
component 이름 수정

Hamthoven committed:
충돌 파일 수정

pbxproj 파일 수정
UIFont+ 파일 수정

Hamthoven committed:
Merge branch 'develop' into feat/#17-split-mypage-profile-p

Hamthoven commented:
@dev-madilyn 충돌 파일은 수정해서 프로젝트 열리고 빌드 된다...

추신... 그리고 네비게이션바 추가하는 건 나중에 할게... 내가 깃을 지금까지 비효율적으로 쓰고 있었다는 걸 지금 깨달았고... 이젠 효율적으로 쓸 수 있을 것 같구나... (다음부턴 네비게이션바처럼 새로 생긴거도 아름답게(?) 적용한 뒤 PR 넣겠다는 이야기)

dev-jungbin commented on SNAPFIT/SNAPFIT/Sources/Components/CarouselCollections/GalleryCollectionViewController.swift:
```diff
@@ -8,7 +8,7 @@
 import SnapKit
 
-class GalleryCollectionlViewController: UIViewController {
+class GalleryCollectionViewController: UIViewController {
```
UICollectionViewController라는 애가 있으니까,
여기서는 GalleryViewController가 좋지 않을까?!
하지만 우리는 진짜 찐 갤러리 뷰가 있기 때문에 차라리 UICollectionView를 상속받고, 우리 디자인상 컬렉션뷰로 갤러리를 보여주는 경우가 많아서, UserGalleryCollectionView 요런식으로 가는 게 어떨까,, 싶습니다아

dev-jungbin commented on SNAPFIT/SNAPFIT/Sources/Components/SnapfitUserInformationViewController.swift:
```diff
@@ -0,0 +1,227 @@
+
+class SnapfitUserInformationViewController: BaseViewController {
+    
+    enum MypageText {
```
Mypage 뷰 안에 있는 Text이기 때문에 그냥 enum Text로 이름을 줘도 괜찮지 않을까??!

다른 프로퍼티들에서 MypageScrollView, MypageContentView라고는 하지 않는 것처럼..!

dev-jungbin reviewed (approved):
굿굿! 코드리뷰 확인 부탁드려요~~ 늦어서 쏘리 🥲

dev-jungbin reviewed (approved):
굿굿! 코드리뷰 확인 부탁드려요~~ 늦어서 쏘리 🥲

Hamthoven committed:
#17 코드리뷰 반영

- UserInformationViewController enum 이름 변경
- GalleryCollectionViewController 이름 UserGalleryCollectionViewController로 변경
